.github/workflows/ci-build.yml

```diff
@@ -0,0 +1,46 @@
+# NOTE: This name appears in GitHub's Checks API and in workflow's status badge.
+name: ci-build
+
+# Trigger the workflow when:
+on:
+  # A push occurs to one of the matched branches.
+  push:
+    branches: [stable, master]
+  # Or when a pull request event occurs for a pull request against one of the
+  # matched branches.
+  pull_request:
+    branches: [stable, master]
+
+# disable secrets.GITHUB_TOKEN permissions
+permissions: {}
+
+jobs:
+  yarn_cache:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js 18.x
+        uses: actions/setup-node@v3
+        id: yarn-cache
+        with:
+          node-version: '18.x'
+          cache: 'yarn'
+      - if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile
+
+  build:
+    # NOTE: This name appears in GitHub's Checks API.
+    name: build
+    needs: [yarn_cache]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
Use `actions/checkout@v4` and add `name:` `Checkout code` to unify this with the https://github.com/oasisprotocol/explorer/blob/master/.github/workflows/ci-build.yml flow.